File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
  def longestCommonPrefix(self, strs: List[str]) -> str:
    trie = {}
    prefix = strs[0]
    for i in range(len(strs)):
      if len(strs[i]) < len(prefix):
        prefix = strs[i]
      node = trie
      for j in range(len(strs[i])):
        if strs[i][j] not in node:
          if i and len(prefix) > j:
            prefix = prefix[:j]
            break
          node[strs[i][j]] = {}
        node = node[strs[i][j]]
    return prefix

        

File: 0014-longest-common-prefix/0014-longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
  prefix := ""
  for j := 0; j < len(strs[0]); j++ {
    for i := 1; i < len(strs); i++ {
      if j == len(strs[i]) || strs[i][j] != strs[0][j] {
        return prefix
      }
    }
    prefix += string(strs[0][j])
  }
  return prefix
}